### PR TITLE
Docker container excessively privileged running root user

### PR DESCRIPTION
**Pull request by zhang-hua**
* Add default non-root user 'app'
* Run service as 'app'
* use ports 10080/10443

PBI[2211778]

# Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [x] All new and existing tests passed.
- [x] The code follows the code style and conventions of this project.
- [ ] The change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

# Description of the change

Without explicitly defining a running user in a Dockerfile definition, the root user is utilized by default.
Each respective Dockerfile should explicitly define a user and mandate that user to be the service runner via the USER instruction. Further, the service user should also have a defined GROUP. Without one, the primary group for the user will fall back to the root group.

# Motivation for the change

...

### CODE REVIEW

File: reverse-proxy/Dockerfile [hunk 1]
```diff
@@ -4,13 +4,21 @@ MAINTAINER Devis Lucato (https://github.com/dluc)
 
 LABEL Tags="Azure,IoT,Solutions,Proxy,nginx"
 
+ARG user=pcsuser
```
Comment (dluc) [inline reverse-proxy/Dockerfile]: ARG user=app

File: reverse-proxy/Dockerfile [hunk 2]
```diff
@@ -4,13 +4,21 @@ MAINTAINER Devis Lucato (https://github.com/dluc)
...
+
+RUN useradd -m -s /bin/bash -U $user
```
[the remark this answers is not in the document]
Comment (zhang-hua) [reply reverse-proxy/Dockerfile]: Looks like command 'groupadd' and 'useradd' are not supported by /bin/sh of BusyBox.

**Comment by ppathan** [comment]
@zhang-hua Have you tested this change with your own docker image in vm?

**Comment by dluc** [comment]
could you update the PR description to match with the changes?

**Comment by zhang-hua** [comment]
@dluc, have updated the PR description.

**Pull request merged**